### PR TITLE
Remove node func callback

### PR DESCRIPTION
**Pull request by k8adev**
- separate log func
- change callback func for async func
- rename some vars and func

### CODE REVIEW

**Comment by k8adev** [comment]
@ErickWendel check this pls

File: src/feeder.ts [hunk 1]
```diff
@@ -52,52 +52,46 @@ export default class Feeder {
...
   }
   /**
    * Check if the output directory exists and
    * create it if it doesn't.
    */
-  async checkOutDir() {
+  async createDir() {
     const dir = path.dirname(this.file);
-    const recursive = true;
+    const hasFile = await fs.existsSync(dir);
```
Comment (ErickWendel) [inline src/feeder.ts]: Fs.existsSync does not return a promise. You'd remove the await inhere

File: src/feeder.ts [hunk 2]
```diff
@@ -52,52 +52,46 @@ export default class Feeder {
...
 
-    if (await fs.existsSync(dir)) {
+    if (hasFile) {
       return;
     }
 
-    await fs.mkdir(
+    await fs.mkdirSync(
```
Comment (ErickWendel) [inline src/feeder.ts]: You youd use fs.promises.mkdir that returns a Promise so you'd use await instead

File: src/feeder.ts [hunk 3]
```diff
@@ -106,38 +100,35 @@ export default class Feeder {
       ext,
     });
 
-    await fs.writeFile(
+    await fs.writeFileSync(
```
Comment (ErickWendel) [inline src/feeder.ts]: Same here. Everything that contains Sync in the name is synchronous so no need to use await. If you do wanna use an asynchronous function prefer fs.promises.writeFile

File: src/index.ts [hunk 4]
```diff
@@ -59,28 +60,25 @@ const feeder = new Feeder(
   { output },
 );
 
-if (argv.parser) {
-  feeder.useParser(argv.parser);
-}
-
 (async () => {
```
Comment (ErickWendel) [inline src/index.ts]: Typescript has top-level await. So you don't need this Immediately Invoked Function Expression (IIFE is (async ()=>{})() ) here

**Review by ErickWendel — CHANGES_REQUESTED** [review]
Good job! We're Almost there! 

You've changed async functions that use callbacks to use sync functions.

We usually use async functions when we'd have many users accessing the same file so the process won't stop until the file contents are ready.

If your feed is a function being executed only by a CLI it's fine but as a matter of studying new things I'd recommend taking a look at the docs on fs.promises module

Also, search about converting callbacks into Promises and why asynchronous functions are useful in JavaScript. This is key for supporting thousands of users concurrently without having to hassle on slow processes

**Comment by k8adev** [comment]
> Good job! We're Almost there!
> 
> You've changed async functions that use callbacks to use sync functions.
> 
> We usually use async functions when we'd have many users accessing the same file so the process won't stop until the file contents are ready.
> 
> If your feed is a function being executed only by a CLI it's fine but as a matter of studying new things I'd recommend taking a look at the docs on fs.promises module
> 
> Also, search about converting callbacks into Promises and why asynchronous functions are useful in JavaScript. This is key for supporting thousands of users concurrently without having to hassle on slow processes

You're awesome, thanks for the observations! 😍

The idea behind using async functions is that the RSS received can have 1 or 10,000 items and may require a custom parser that this library cannot control. Therefore, I think promises can be useful in this case. What do you think about it?

*I'm bringing here some of the contents that you indicated to me earlier as a note*:

**Como Converter Callbacks em Promises em Javascript**
https://www.youtube.com/watch?v=_gmXKsVS40s

**Javascript Callbacks, Promises e Async/Await**
https://www.maiconsilva.com/js-callbacks-promises-aw/

**Pull request merged**